<launch>
  <include file="$(find xr_tank_moveit)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <rosparam file="$(find xr_tank)/config/xr_tank_control.yaml" command="load"/>

  <!-- controller manager & hardware interface -->
  <node name="xr_tank_hw" pkg="xr_tank" type="xr_tank_hw" output="screen" ns="xr_tank">
  </node>

  <!-- ros controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
	output="screen" ns="xr_tank" args="joint_state_controller
    arm_controller gripper_l_position_controller
    cam_pan_position_controller cam_tilt_position_controller
    wheel_l_effort_controller wheel_r_effort_controller"/>

  <!-- moveit arm control -->
  <include file="$(find xr_tank_moveit)/launch/move_group.launch">
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="fake_execution" value="false"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="false"/>
  </include>

  <node name="arm_control_node" pkg="xr_tank" type="arm_control_node.py" output="screen">
    <remap from="/joint_states" to="/xr_tank/joint_states"/>
  </node>


  <!--car control-->
  <node name="cam_capture_node" pkg="xr_tank" type="cam_capture_node.py" output="screen">
      <param name="camID" value="0" />
      <param name="flipH" value="false" />
      <param name="flipV" value="false" />
  </node>
  <node name="car_control_node" pkg="xr_tank" type="car_control_node.py" output="screen">
      <param name="leftRatio" value="1" />
      <param name="rightRatio" value="1" />
  </node>

  <node name="xr_driver_node" pkg="xr_tank" type="xr_driver_node.py" output="screen">
  </node>

  <!--<node name="yolo_node" pkg="xr_tank" type="yolo_node" output="screen"></node>


  <include file="$(find xr_tank)/launch/joy_control.launch"></include>-->

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
  respawn="false" output="screen">
    <remap from="/joint_states" to="/xr_tank/joint_states"/>
  </node>
  
  <!--<node name="rviz" pkg="rviz" type="rviz" args="-d $(find xr_tank)/config/xr_tank_moveit.rviz"/>-->


</launch>
